src/os/bluestore/BlueStore.cc

```diff
@@ -1670,7 +1670,7 @@ void BlueStore::SharedBlob::put()
 			     << " removing self from set " << get_parent()
 			     << dendl;
     if (get_parent()) {
-      if (get_parent()->remove(this)) {
+      if (get_parent()->try_remove(this)) {
```

Comment:
Funny thing but you can use NEW remove here instead and get rid off try_remove at all. See if (--nrerf == 0) above...

Reply:
We don't take the lock until inside try_remove(), and we're handling the race between one thread dropping the last reference and trying to remove from the man and another thread calling lookup() and bumping the ref from 0 -> 1.